generative_data_prep/utils/logger.py

```diff
@@ -0,0 +1,125 @@
+"""Copyright 2023 SambaNova Systems, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+
+This class creates a common logger.
+"""
+import datetime
+import logging
+import logging.config
+import os
+from pathlib import Path
+
+import git
+import yaml
+
+try:
+    SEP_STR = "-" * os.get_terminal_size().columns
+except OSError:
+    SEP_STR = "-" * 80
+
+
+class Logger(object):
```

Comment:
is there a reason why we're not using logger as specified by the official Python docs?

We can simply create namespaces under each module when importing `logger` and `logger` is implicitly already treated as singleton in Python.
- https://docs.python.org/3/howto/logging.html#advanced-logging-tutorial

A more distilled explanation with practical examples is here:
- https://stackoverflow.com/a/15735146/2868465

Reply:
Done!